### PR TITLE
Permit UUID by default

### LINKED ISSUE #217: DefaultInvocationSerializer does not allow java.util.UUID
`java.util.UUID` is a commonly-used JDK class and is commonly used to model surrogate identifiers. It is therefore a good candidate to be passed as an argument to "scheduled" invocations with transaction outbox.

`DefaultInvocationSerializer` does not currently appear to allow it. There are various work-arounds, but it seems like something that should be allowed.  Do you agree? If so I will raise a PR.

### PR DESCRIPTION
**Pull request by tsg21**
Make `DefaultInvocationSerializer` permit `java.util.UUID`

Fixes #217

### CODE REVIEW

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java [hunk 1]
```diff
@@ -125,41 +128,44 @@ public Invocation deserializeInvocation(Reader reader) {
       addClassPair(boolean.class, "boolean");
       addClassPair(char.class, "char");
 
-      addClassPair(Byte.class, "Byte");
-      addClassPair(Short.class, "Short");
-      addClassPair(Integer.class, "Integer");
-      addClassPair(Long.class, "Long");
-      addClassPair(Float.class, "Float");
-      addClassPair(Float.class, "Float");
-      addClassPair(Float.class, "Float");
```
Comment (tsg21) [inline transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:134]: Note the duplication here, this has been removed.

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java [hunk 2]
```diff
@@ -125,41 +128,44 @@ public Invocation deserializeInvocation(Reader reader) {
...
+      addClass(MonthDay.class);
+      addClass(Period.class);
+      addClass(Year.class);
+      addClass(YearMonth.class);
+      addClass(ZoneOffset.class);
+      addClass(DayOfWeek.class);
+      addClass(ChronoUnit.class);
+
+      addClass(Transaction.class);
+      addClass(TransactionContextPlaceholder.class);
```
Comment (amseager) [inline transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:160]: That differs from `addClassPair(TransactionContextPlaceholder.class, "TransactionContext");`, is it ok?

Comment (tsg21) [reply transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:160]: Good spot! I went through them and failed on the last line. I will undo that - I don't have an interest in accidentally altering behaviour here - the old name might be persisted.

Comment (tsg21) [reply transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java]: Fixed. Reverted to the old name for this one.

**Comment by badgerwithagun** [comment]
Thanks @tsg21. Merged :)

**Pull request merged**